### DISCUSSION TITLE
Failing to create proposal through Snapshot.js

### DISCUSSION THREAD

**Discussion by jamesdobry** [snapshot-labs/snapshot#1772] [Q&A] [answered]
Hello,

I am trying to integrate Snapshot into my dApp and am struggling with a few things since the documentation is very sparse. 

When I go to create a proposal, I have no idea what to set the `snapshot` variable to. Also, what is allowed to be dumped to the `metadata` variable? 

The main issue right now is when I go to publish, I get the following error:

```
Error: overflow (fault="overflow", operation="BigNumber.from", value={LONG NUMBER}, code=NUMERIC_FAULT, version=bignumber/5.5.0)
```

Does anyone have any ideas? I am passed in the wallet address into the second parameter and I am not setting the `from` variable.

Thanks for the help,

James

**Comment by metrox-eth** [comment]
I think I can answer for the first part, `snapshot` is the block number the closer to the creation of the proposal.
Not sure if metadata is mandatory.
@ChaituVR  Maybe you can help ?

**Comment by ChaituVR** [comment, marked as answer]
@jamesdobry Hello, 

`snapshot`  is the block number of the network, ideally, pass the latest block number while creating a proposal
`metadata` can be ignored
Not sure what is the error is about, is your code open-source? 

This code sandbox https://codesandbox.io/s/vibrant-dawn-u7r43 might be helpful

**Reply by jamesdobry** [reply]
Thank you! So the error that was being thrown was due to my start/end timestamp calculations were being off.

**Reply by LucasGrasso** [reply]
@jamesdobry @ChaituVR  Greetings!

I am running through the same issue, in what format should end and start be?

```js
const date = Date.now();
const blockNumber = await web3.eth.getBlockNumber();
const isValidProposal = snapshot.utils.validateSchema(
snapshot.schemas.proposal,
{
	space: "zerti.eth",
	type: "basic",
	choices: ["For", "Against", "Abstain"],
	title: `Should 0x00 be added to the Zerti DAO as an entity?`,
	body: "test",
	start: date,
	end: date + 432000000, 
	snapshot: blockNumber,
	network: "5",
}
	);
	console.log('Proposal', isValidProposal);
```

**Reply by ChaituVR** [reply]
@LucasGrasso What error do you see? compare these values with the snapshot UI when you create a proposal on your space

**Comment by Kumarmohang** [comment]
import "./App.css";
import { Web3Provider } from "@ethersproject/providers";
import snapshot from "@snapshot-labs/snapshot.js";

function App() {
  //const hub = "https://hub.snapshot.org"; // or https://testnet.snapshot.org for testnet

  const hub = "https://testnet.snapshot.org";
  const client = new snapshot.Client712(hub);

  const Proposal = async () => {
    const web3 = new Web3Provider(window.ethereum);
    const [account] = await web3.listAccounts();

    console.log({ account });

    const blockNumber = await web3.getBlockNumber();
    console.log({ blockNumber });

    const receipt = await client.proposal(web3, account, {
      space: "yam.eth",
      type: "single-choice", // define the voting system
      title: "Test proposal using Snapshot.js",
      body: "This is the content of the proposal",
      choices: ["For", "Against", "Abstain"],
      start: 1679565857,
      end: 1637244000,
      snapshot: blockNumber,
      network: "5",
      plugins: JSON.stringify({}),
      app: "my-app", // provide the name of your project which is using this snapshot.js integration
    });

    console.log({ receipt });
  };

  return (
    <div className="App">
      <header className="App-header">
        <button className="btn" onClick={Proposal}>
          {" "}
          <h1>Create Proposal</h1>
        </button>
      </header>
    </div>
  );
}

export default App;




i am above code for the proposal creation . its giving wrong proposal format . kindly help
